packages/salesforcedx-vscode-core/src/commands/forceOrgOpen.ts

```diff
@@ -6,24 +6,64 @@
  */
 
 import {
+  CliCommandExecutor,
   Command,
   SfdxCommandBuilder
 } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
 import { nls } from '../messages';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath } from '../util';
 import {
   EmptyParametersGatherer,
   SfdxCommandlet,
   SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from './commands';
 
+interface CLIOrgData {
+  status: number;
+  result: {
+    orgId: string;
+    url: string;
+    username: string;
+  };
+}
+
 class ForceOrgOpenExecutor extends SfdxCommandletExecutor<{}> {
   public build(data: {}): Command {
-    return new SfdxCommandBuilder()
+    const builder = new SfdxCommandBuilder()
       .withDescription(nls.localize('force_org_open_default_scratch_org_text'))
       .withArg('force:org:open')
-      .withLogName('force_org_open_default_scratch_org')
-      .build();
+      .withLogName('force_org_open_default_scratch_org');
+    if (process.env.SFDX_CONTAINER_MODE) {
+      builder.withArg('-r').withJson();
+    }
+    return builder.build();
+  }
+
+  public execute(response: ContinueResponse<string>): void {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    if (process.env.SFDX_CONTAINER_MODE) {
```

Comment:
I wonder if we should use this (which we set by ourselves) or if we should use a more general one from VS Code itself to detect if it is running in container mode.

What's the status of the general one from VS Code?

Reply:
oh, forgot there is a general one from VS code, I'll have to look into that.
Is there an advantage to using our own here when we start running the Web IDE, will can we set the same env var?

Reply:
We decided against using the vscode one so that it will work in the web-ide container.

Reply:
Correct me if I am wrong, we decided on using a ENV VARIABLE so that the CLI can detect it and use the device flow for that.

But, I don't think it's mutually exclusive right? When we are in VS Code, we could use its own detection mechanism?

Reply:
What is advantage to using VS Codes detection system over our own ENV var? The assumption is its more reliable or that if a user didn't start with our supported image then the container adaptations would be broken again?

Reply:
What happens if a user is not using our official docker image and doesn't set the ENV var?

Reply:
Then the command would proceed as if its not in a container and try to open a browser, which may not be available in the container.